pyquil/api/_base_connection.py

```diff
@@ -260,6 +261,217 @@ def qvm_run_payload(quil_program, classical_addresses, trials,
     return payload
 
 
+class ForestSession(requests.Session):
+    """
+    ForestSession inherits from requests.Session. It is responsible for adding
+    authentication headers to Forest server requests. Upon receiving a 401 or 403
+    response, it will attempt to refresh the auth credential and update the
+    PyquilConfig, which in turn writes the refreshed auth credential to file.
+
+    Encapsulates the operations required for authorization & encryption
+      with the QPU.
```

Comment:
I don't have strong opinions about docstring formatting, but I do have a preference for consistency. Most of the existing docstrings don't indent non-head lines in a paragraph as such, so probably we shouldn't do it here (or for that matter in the other dozen or so instances which follow).